public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx

```diff
@@ -151,27 +155,163 @@ export const ConnectionSVG = ({
+                ya = (lDelta + 2 * lHalfArc) * Math.sin(angle1) + y1;
+                xb = -lHalfArc * Math.cos(angle2) + X;
+                yb = -lHalfArc * Math.sin(angle2) + Y;
+              }
+            }
           } else {
             // For all other vertices
             const previousVertex = vertices[index - 1];
             addVertices.push(calculateMidpoint(previousVertex.x, previousVertex.y, x, y));
+
+            // Only calculate arcs if there is a radius
+            if (radius) {
+              // Convert previous vertex relative coorindates to scene coordinates
+              const Xp = previousVertex.x * xDist + x1;
+              const Yp = previousVertex.y * yDist + y1;
+
+              // Length of segment
+              const lSegment = calculateDistance(X, Y, Xp, Yp);
+              if (Math.abs(lHalfArc) > 0.5 * Math.abs(lSegment)) {
+                // Limit curve control points to mid segment
+                lHalfArc = 0.5 * lSegment;
+              }
+              // Default next point to last point
+              let Xn = x2;
+              let Yn = y2;
+              if (index < vertices.length - 1) {
+                // Not also the last point
+                const nextVertex = vertices[index + 1];
+                Xn = nextVertex.x * xDist + x1;
+                Yn = nextVertex.y * yDist + y1;
+              }
+
+              // Length of next segment
+              const lSegmentNext = calculateDistance(X, Y, Xn, Yn);
+              if (Math.abs(lHalfArc) > 0.5 * Math.abs(lSegmentNext)) {
+                // Limit curve control points to mid segment
+                lHalfArc = 0.5 * lSegmentNext;
+              }
+
+              // Calculate arc control points
+              const lDelta = lSegment - lHalfArc;
+              xa = lDelta * Math.cos(angle1) + Xp;
+              ya = lDelta * Math.sin(angle1) + Yp;
+              xb = lHalfArc * Math.cos(angle2) + X;
+              yb = lHalfArc * Math.sin(angle2) + Y;
+
+              // Check if arc control points are inside of segment, otherwise swap sign
+              if ((xa > X && xa > Xp) || (xa < X && xa < Xp)) {
+                xa = (lDelta + 2 * lHalfArc) * Math.cos(angle1) + Xp;
+                ya = (lDelta + 2 * lHalfArc) * Math.sin(angle1) + Yp;
+                xb = -lHalfArc * Math.cos(angle2) + X;
+                yb = -lHalfArc * Math.sin(angle2) + Y;
+              }
+            }
           }
           if (index === vertices.length - 1) {
-            // For last vertex
+            // For last vertex only
             addVertices.push(calculateMidpoint(1, 1, x, y));
           }
+          // Add segment to path
+          pathString += `L${xa} ${ya} `;
+
+          if (lHalfArc !== 0) {
+            // Add arc if applicable
+            pathString += `Q ${X} ${Y} ${xb} ${yb} `;
+          }
         });
+        // Add last segment
         pathString += `L${x2} ${y2}`;
```

Comment:
minor for later but ability to draw vertex over actual line / some way of indicating relation or something
![Screenshot 2024-03-22 at 11 23 40 AM](https://github.com/grafana/grafana/assets/22381771/7dc7fe9d-1f6f-4f1a-9688-c6855c004cdd)